### PR TITLE
improve setValue and watch type

### PR DESCRIPTION
**Pull request by kotarella1110**

### CODE REVIEW

File: src/types/contextTypes.ts [hunk 1]
```diff
@@ -68,15 +66,14 @@ export type UseFormMethods<TFieldValues extends FieldValues = FieldValues> = {
...
+  >(
     name: TFieldName,
-    value: TFieldName extends keyof TFieldValues
-      ? IsAny<TFieldValues[TFieldName]> extends true
-        ? any
-        : TFieldValues[TFieldName] extends NestedValue<infer U>
-        ? U
-        : UnpackNestedValue<DeepPartial<TFieldValues[TFieldName]>>
-      : LiteralToPrimitive<TFieldValue>,
+    value: TFieldValue extends NestedValue<infer U>
```
Comment (bluebill1049) [inline src/types/contextTypes.ts:74]: if the field value, matches nestedValue's value return it's type?

Comment (kotarella1110) [reply src/types/contextTypes.ts:75]: @bluebill1049 what do you mean?

```ts
const { setValue } = useForm<{ test: NestedValue<{ key: string }> }>();

setValue('test', { key: 'value' });
```

Comment (bluebill1049) [reply src/types/contextTypes.ts:75]: oh sorry, I read my own sentence again and it's confusing... 

so the type was meant to infer the type inside NestedValue right?

Comment (kotarella1110) [reply src/types/contextTypes.ts:75]: @bluebill1049 yes. you're completely right!

**Pull request merged**